platform/wayland/cog-platform-wl.c

```diff
@@ -271,8 +271,14 @@ pointer_on_enter(void              *data,
 static void
 pointer_on_leave(void *data, struct wl_pointer *pointer, uint32_t serial, struct wl_surface *surface)
 {
+    if (data == NULL || pointer == NULL)
```

Comment:
In general it's not needed to explicitly compare with `NULL`, which tends to be clearer:

```suggestion
    if (!data || !pointer)
```